src/jit/lowerxarch.cpp

```diff
@@ -2680,9 +2680,17 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* containingNode, Ge
             assert(supportsUnalignedSIMDLoads == false);
             assert(supportsSIMDScalarLoads == false);
 
-            const unsigned expectedSize = genTypeSize(containingNode->TypeGet());
+            unsigned       expectedSize = genTypeSize(containingNode->TypeGet());
             const unsigned operandSize  = genTypeSize(node->TypeGet());
 
+            // CRC32 codegen depends on its second oprand's type.
+            // Currently, we are using SIMDBaseType to store the op2Type info.
+            if (containingIntrinsicId == NI_SSE42_Crc32)
+            {
+                var_types op2Type = containingNode->gtSIMDBaseType;
+                expectedSize      = genTypeSize(op2Type);
+            }
```

Comment:
Also fixed an issue that prevents `Crc32` containment on `byte` and `ushort`.

Reply:
Do we have any other scalar intrinsics with similar requirements?

Reply:
No, Crc32 is the only one.